core/src/rpc_subscriptions.rs

```diff
@@ -11,17 +11,40 @@ use solana_sdk::{
 };
 use solana_vote_program::vote_state::MAX_LOCKOUT_HISTORY;
 use std::collections::HashMap;
-use std::sync::{Arc, RwLock};
+use std::ops::DerefMut;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{Receiver, RecvTimeoutError, SendError, Sender};
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread::{Builder, JoinHandle};
+use std::time::Duration;
+
+const RECEIVE_DELAY_MILLIS: u64 = 100;
 
 pub type Confirmations = usize;
 
-#[derive(Serialize, Clone)]
+#[derive(Serialize, Clone, Copy, Debug)]
 pub struct SlotInfo {
     pub slot: Slot,
     pub parent: Slot,
     pub root: Slot,
 }
 
+pub enum NotificationEntry {
+    Slot(SlotInfo),
+    Bank((Slot, Arc<RwLock<BankForks>>)),
+}
+
```

Comment:
from @mvines:

![image](https://user-images.githubusercontent.com/153621/72641186-85dc1900-3937-11ea-8e8b-9fe11f945a8c.png)

Reply:
`Arc<T>` alone was not enough, needed an `Arc<Mutex<T>>` - see `channel()` creation below